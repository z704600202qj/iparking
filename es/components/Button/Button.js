import * as React from 'react';
import noop from '../../utils/noop';

var Button = function Button(props) {
  var _props$onClick = props.onClick,
      onClick = _props$onClick === void 0 ? noop : _props$onClick,
      children = props.children;

  var handleClick = function handleClick(event) {
    onClick(event);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "button",
    role: "button",
    onClick: handleClick
  }, children);
};

export default Button;