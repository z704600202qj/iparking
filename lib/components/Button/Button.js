"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _noop = _interopRequireDefault(require("../../utils/noop"));

var Button = function Button(props) {
  var _props$onClick = props.onClick,
      onClick = _props$onClick === void 0 ? _noop["default"] : _props$onClick,
      children = props.children;

  var handleClick = function handleClick(event) {
    onClick(event);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "button",
    role: "button",
    onClick: handleClick
  }, children);
};

var _default = Button;
exports["default"] = _default;