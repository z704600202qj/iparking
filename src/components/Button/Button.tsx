import * as React from 'react';
import noop from '../../utils/noop';

export interface ButtonProps {
  type?: 'primary' | 'ghost' | 'warning' | 'default';
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
  loading?: boolean;
}

const Button: React.FC<ButtonProps> = (props) => {
  const {
    onClick = noop,
    children,
  } = props;

  const handleClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>): void => {
    onClick(event);
  };

  return (
    <div
      className="button"
      role="button"
      onClick={handleClick}
    >
      {
        children
      }
    </div>
  );
};

export default Button;
