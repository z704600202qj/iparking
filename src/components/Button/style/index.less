@import "~src/styles/index.less";
@import "./common.less";