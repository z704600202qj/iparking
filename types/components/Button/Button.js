import * as React from 'react';
import noop from '../../utils/noop';
const Button = (props) => {
    const { onClick = noop, children, } = props;
    const handleClick = (event) => {
        onClick(event);
    };
    return (React.createElement("div", { className: "button", role: "button", onClick: handleClick }, children));
};
export default Button;
