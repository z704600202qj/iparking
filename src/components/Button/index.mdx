---
name: Button
route: /Button
menu: 组件
---

import { Playground } from 'docz'; 
import Button from './index'

警告提示，展现需要关注的信息。

## 代码演示

### 基本用法

<Playground>
  <Button>34</Button>
</Playground>

## API

| 属性 | 说明     | 类型                                         | 默认值 |
| ---- | -------- | -------------------------------------------- | ------ |
| kind | 警告类型 | 'info'/'positive'/'negative'/'warning'非必填 | 'info' |