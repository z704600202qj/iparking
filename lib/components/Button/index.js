"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _Button = _interopRequireDefault(require("./Button"));

require("./style");

var _default = _Button["default"];
exports["default"] = _default;