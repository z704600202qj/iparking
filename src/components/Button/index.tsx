import Button from './Button';
import './style';

export default Button;
