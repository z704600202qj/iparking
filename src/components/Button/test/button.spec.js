import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../Button';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {

  it('Button组件快照测试', () => {
    const component = renderer.create(
      <Button>click</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('测试Button组件的children渲染', () => {
    const wrapper = Enzyme.shallow(
      <Button>click</Button>
    );
    expect(wrapper.contains('click')).toBe(true);
  });

  it('模拟Button组件的点击回调', () => {
    let count = 0;
    const handleClick = () => {
      count += 1;
    };
    const wrapper = Enzyme.shallow(
      <Button onClick={handleClick}/>
    );
    wrapper.find('div').simulate('click');
    expect(count).toBe(1);
  });

  it('测试Button组件的DOM结构', () => {
    const wrapper = Enzyme.mount(
      <Button>Hello</Button>
    );
    expect(wrapper.html()).toBe('<div class="button" role="button">Hello</div>');
  });
});
