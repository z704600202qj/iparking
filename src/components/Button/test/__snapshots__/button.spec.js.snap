// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button组件快照测试 1`] = `
<div
  className="button"
  onClick={[Function]}
  role="button"
>
  click
</div>
`;
